components/Swap/Withdraw/WalletStore.ts

```diff
@@ -0,0 +1,115 @@
+import { create } from 'zustand'
+import { SwapItem } from '../../../lib/layerSwapApiClient'
+import { CryptoNetwork, NetworkType } from '../../../Models/CryptoNetwork'
+import { GetDefaultNetwork } from '../../../helpers/settingsHelper'
+import { StarknetWindowObject } from 'get-starknet'
+import KnownInternalNames from '../../../lib/knownIds'
+import { Layer } from '../../../Models/Layer'
+import { createPublicClient, http, parseAbi } from 'viem'
+import resolveChain from '../../../lib/resolveChain'
+
+type Networks = {
+    [network: string]: Network
+}
+
+export type Network = {
+    address: string,
+    metadata?: Metadata
+}
+
+export type Metadata = {
+    isArgent?: boolean
+}
+
+interface WalletState {
+    networks: Networks
+    addNetwork: (address: string, source_layer: Layer, swap?: SwapItem, starknetAccount?: StarknetWindowObject, imxAccount?: string) => void
+    updateNetworkMetadata: (swap: SwapItem, metadata: Metadata) => void
+}
+
+export const useWalletStore = create<WalletState>()((set) => ({
+    networks: {},
+    addNetwork: async (address, source_layer, swap, starknetAccount, imxAccount) => {
+
+        if (!source_layer) return
+
+        let isArgent: boolean
+        let accountAddress: string
+
+        // Check if addres generated by Argent Wallet
+        if (source_layer?.internal_name === KnownInternalNames.Networks.EthereumMainnet) {
+            isArgent = await isArgentWallet(address, source_layer?.assets?.[0].network)
+        }
+
+        // Set source address
```

Comment:
This whole block is bad

Reply:
You should have a getter, that will provide the correct address based on the provided network